xclim/indices/_multivariate.py

```diff
@@ -479,6 +481,62 @@ def heat_wave_max_length(
     return max_l.where(max_l >= window, 0)
 
 
+@declare_units(
+    "days",
+    tasmin="[temperature]",
+    tasmax="[temperature]",
+    thresh_tasmin="[temperature]",
+    thresh_tasmax="[temperature]",
+)
+def heat_wave_total_length(
+    tasmin: xr.DataArray,
+    tasmax: xr.DataArray,
+    thresh_tasmin: str = "22.0 degC",
+    thresh_tasmax: str = "30 degC",
+    window: int = 3,
+    freq: Optional[str] = None,
+) -> xr.DataArray:
+    # Dev note : we should decide if it is deg K or C
+    r"""Heat wave total length
+
+    Total length of heat waves over a given period. A heat wave is defined as an event
+    where the minimum and maximum daily temperature both exceeds specific thresholds
+    over a minimum number of days. This the sum of all days in such events.
+
+    Parameters
+    ----------
+    tasmin : xr.DataArray
+      Minimum daily temperature [℃] or [K]
+    tasmax : xr.DataArray
+      Maximum daily temperature [℃] or [K]
+    thresh_tasmin : str
+      The minimum temperature threshold needed to trigger a heatwave event [℃] or [K]. Default : '22 degC'
+    thresh_tasmax : str
+      The maximum temperature threshold needed to trigger a heatwave event [℃] or [K]. Default : '30 degC'
+    window : int
+      Minimum number of days with temperatures above thresholds to qualify as a heatwave.
+    freq : Optional[str]
+      Resampling frequency; Defaults to "YS".
+
+    Returns
+    -------
+    xr.DataArray
+      Total length of heatwave at the wanted frequency
+
+    Notes
+    -----
+    See notes and references of `heat_wave_max_length`
+    """
+    freq = freq or "YS"
+    thresh_tasmax = utils.convert_units_to(thresh_tasmax, tasmax)
+    thresh_tasmin = utils.convert_units_to(thresh_tasmin, tasmin)
+
+    cond = (tasmin > thresh_tasmin) & (tasmax > thresh_tasmax)
+    group = cond.resample(time=freq)
+    total_count = group.apply(lambda d, dim=None: rl.windowed_run_count(d, window, dim=dim), dim="time")
```

Comment:
Why are you passing `dim=None` ?

I think `group.apply(rl.windowed_run_count, window=window, dim="time")` would work as well.

Reply:
This was my same question...  the apply call doesn't seem as 'clean' as heat wave max length etc.  Are there advantages to using a lambda func here?

Reply:
Yeah... I wasn't sure how to do it, I think my new commit fixes it.

Reply:
I thin in this case we could simply do:

```return group.apply(rl.windowed_run_count, window=window, dim="time")```